src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -569,6 +584,114 @@ FullQName AdvertiserMinMdns::GetCommisioningTextEntries(const CommissionAdvertis
     }
 }
 
+bool CanAdvertiseOn(chip::Inet::InterfaceAddressIterator & current)
+{
+    if (!current.IsUp())
+    {
+        return false;
+    }
+
+    if (!current.SupportsMulticast())
+    {
+        return false;
+    }
+
+    char name[chip::Inet::InterfaceIterator::kMaxIfNameLength];
+    if (current.GetInterfaceName(name, sizeof(name)) != CHIP_NO_ERROR)
+    {
+        ChipLogError(Discovery, "Failed to get interface name - will not advertise on this interface");
+        return false;
+    }
+
+    if (strncmp(name, "lo", 2) == 0)
+    {
+        // Skip advertisement on what looks like a local loopback interface
+        return false;
+    }
+
+    return true;
+}
+
+bool AdvertiserMinMdns::ShouldAdvertiseOn(const chip::Inet::InterfaceId id, const chip::Inet::IPAddress & addr)
+{
+    for (unsigned i = 0; i < mServer.GetEndpointCount(); i++)
+    {
+        const ServerBase::EndpointInfo & info = mServer.GetEndpoints()[i];
+
+        if (info.udp == nullptr)
+        {
+            continue;
+        }
+
+        if (info.interfaceId != id)
+        {
+            continue;
+        }
+
+        if (info.addressType != addr.Type())
+        {
+            continue;
+        }
+
+        return true;
+    }
+
+    return false;
+}
+
+void AdvertiserMinMdns::AdvertiseRecords()
+{
+    chip::Inet::InterfaceAddressIterator interfaceAddress;
+
+    if (!interfaceAddress.Next())
+    {
+        return;
+    }
+
+    for (; interfaceAddress.HasCurrent(); interfaceAddress.Next())
+    {
+        if (!CanAdvertiseOn(interfaceAddress))
+        {
+            continue;
+        }
+
+        if (!ShouldAdvertiseOn(interfaceAddress.GetInterfaceId(), interfaceAddress.GetAddress()))
+        {
+            continue;
+        }
+
+        chip::Inet::IPPacketInfo packetInfo;
+
+        packetInfo.Clear();
+        packetInfo.SrcAddress = interfaceAddress.GetAddress();
+        if (interfaceAddress.GetAddress().IsIPv4())
+        {
+            chip::Inet::IPAddress::FromString("224.0.0.251", packetInfo.DestAddress);
+        }
+        else
+        {
+            chip::Inet::IPAddress::FromString("FF02::FB", packetInfo.DestAddress);
```

Comment:
Worth adding a comment that this is the mDNS multicast address.  Or defining some constants, like `kMdnsPort`, for these strings.

Reply:
Removed some duplication for this and comment on the class hopefully works.